3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,56 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
+async function fetchData(url) {
   // TODO complete this function
+  try{
+    const fetching= await fetch(url)
+  const respond =await fetching.json()
+  return respond
+  }
+  catch(error){
+    console.log(error);
+
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+function fetchAndPopulatePokemons() {
   // TODO complete this function
+  const selectPokemon = document.createElement("select")
+    document.body.appendChild(selectPokemon)
+   fetchData("https://pokeapi.co/api/v2/pokemon/")
+   .then((data)=>{
+    data.results.forEach(el => {
+      const optionElement=document.createElement("option")
+      optionElement.text=el.name
+      selectPokemon.add(optionElement)
+      optionElement.value=el.name
+      return data
+    })
+  })
+  return selectPokemon
 }
-
-function fetchImage(/* TODO parameter(s) go here */) {
+function fetchImage() {
   // TODO complete this function
+  const selectImage=document.createElement("img")
+  document.querySelector("select").onchange=()=>{
+      fetchData(`https://pokeapi.co/api/v2/pokemon/${ document.querySelector("select").value}`)
+      .then((data)=>{
+        selectImage.src=data.sprites.front_default
+        document.body.appendChild(selectImage)
+        return data
+      })
+      .catch((error)=>{
+        console.log(error)
+      })
+  }
 }
 
+
 function main() {
   // TODO complete this function
+  fetchAndPopulatePokemons()
+fetchImage()
```

Comment:
Pay a bit more attention to the formatting. Make sure that the `fetchImage()` and `fetchAndPopulatePokemons()` are aligned in the main function